File: result/translate/starchat/python2java/java/NUMBER_N_DIGIT_STEPPING_NUMBERS.java
public static int f_gold ( int n ) {
        int[][] dp = new int[n + 1][10];
        for (int i = 0; i < 10; i++) {
            dp[1][i] = 1;
        }
        for (int i = 2; i <= n; i++) {
            for (int j = 0; j < 10; j++) {
                if (j == 0) {
                    dp[i][j] = dp[i - 1][j + 1];
                } else if (j == 9) {
                    dp[i][j] = dp[i - 1][j - 1];
                } else {
                    dp[i][j] = (dp[i - 1][j - 1] + dp[i - 1][j + 1]);
                }
            }
        }
        int sum = 0;
        for (int j = 0; j < 10; j++) {
            sum = sum + dp[n][j];
        }
        return sum;
    }

File: result/translate/starchat/java2python/python/NUMBER_N_DIGIT_STEPPING_NUMBERS.py
def f_gold(n):
        dp = [[0] * 10 for _ in range(n + 1)]
        for j in range(10):
            dp[1][j] = 1
        for i in range(2, n + 1):
            for j in range(10):
                if j == 0:
                    dp[i][j] = dp[i - 1][j + 1]
                elif j == 9:
                    dp[i][j] = dp[i - 1][j - 1]
                else:
                    dp[i][j] = dp[i - 1][j - 1] + dp[i - 1][j + 1]
        sum = 0
        for j in range(1, 10):
            sum += dp[n][j]
        return sum